org.eclipse.scanning.points/src/org/eclipse/scanning/points/validation/ScanRequestValidator.java

```diff
@@ -83,16 +86,15 @@ private void validateAnnotations(Map<String, Object> dmodels) throws ValidationE
 						FieldDescriptor des = (FieldDescriptor)annotation;
 						if (des.device()==DeviceType.RUNNABLE) { // Then its value must be in the devices.
 							
-							boolean accessible = field.isAccessible();
-							try {
-								field.setAccessible(true);
-								String value = (String)field.get(model);
-								if (!dmodels.containsKey(value)) {
-									String label = des.label()!=null && des.label().length()>0 ? des.label() : field.getName();
-									throw new ModelValidationException("The value of '"+label+"' references a device ("+value+") not in the scan!", model, field.getName());
+							if (beanMap == null) beanMap = new BeanMap(model);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00122)